### PR TITLE
feat(message-parser): horizontal rule (thematic break)

### PR DESCRIPTION
**Pull request by ggazzo**
## Proposed changes (including videos or screenshots)

Adds horizontal rules (thematic breaks) to the message parser and renderer.

A line consisting of **3 or more contiguous** `-`, `*`, or `_` markers — with nothing else on the line (leading/trailing spaces allowed) — is parsed into a new `HORIZONTAL_RULE` block node:

```
above
---
below
```

- `---`, `***`, `___`, `----------` all produce a rule.
- Rendered with Fuselage's `Divider` in `gazzodown`.
- New `HORIZONTAL_RULE` AST node: `{ type: 'HORIZONTAL_RULE', value: undefined }`.

### Why contiguous-only

Spaced variants like `* * *` and `_ _ _` are **intentionally not matched**. Rocket.Chat already treats `** **` / `__ __` as whitespace-emphasis (plain text); matching spaced markers as a rule would regress that behavior. Contiguous-only keeps the common `---` / `***` / `___` forms working without touching emphasis.

## Issue(s)

Part of the message-parser markdown gap work (thematic break / P1).

## Steps to test or reproduce

1. `cd packages/message-parser && yarn jest tests/horizontalRule.test.ts`
2. Visual: `yarn workspace @rocket.chat/gazzodown storybook` → `Markup` stories → `HorizontalRule` / `Example`.

Or paste in a message:

```
section one
---
section two
```

## Further comments

Full parser suite green (645 tests). Spaced thematic breaks (`* * *`) can be revisited later if the emphasis grammar is reworked.

<a href="https://cubic.dev/pr/RocketChat/Rocket.Chat/pull/41113?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>
 

 Task: [ARCH-2198]

[ARCH-2198]: https://rocketchat.atlassian.net/browse/ARCH-2198?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **New Features**
  * Added horizontal-rule support in message parsing and rendering, displaying a divider between sections.
  * Recognizes lines containing 3+ contiguous `-` characters (allowing optional surrounding spaces).
  * Added a Storybook story and updated examples to demonstrate horizontal-rule rendering.
* **Bug Fixes**
  * Ensures `*`/`_`-style marker-only lines are not interpreted as horizontal rules, preserving existing emphasis and masking behavior.
  * Includes an optional source fallback for renderers that can’t handle horizontal rules directly.
* **Tests**
  * Added comprehensive horizontal-rule parser test coverage, including correct source offsets and non-matching edge cases.

### CODE REVIEW

File: packages/message-parser/src/definitions.ts [hunk 1]
```diff
@@ -123,6 +123,17 @@ export type LineBreak = {
...
 
+// [start, end] offsets into the original parsed source. Consumers that don't
+// implement a renderer for a block can slice the source to show the raw markup,
+// without duplicating the text into the AST.
+export type SourceRange = [number, number];
+
+export type HorizontalRule = {
+	type: 'HORIZONTAL_RULE';
+	value: undefined;
+	fallback?: SourceRange;
```
Comment (sampaiodiego) [inline packages/message-parser/src/definitions.ts]: `fallback` here is `[start, end]` offsets, but `Timestamp` already has a `fallback` field that is a `Plain` node — same field name, two different shapes.

should both be combined or should this one have a different name?

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

**Review by juliajforesti — DISMISSED** [review]
